Latest changes in "dist" break compilation on TS 2.8.4

Seems that latest commit 6df524d388d183d9ce79fe58a50152eff4663a74 in the dist branch brakes the project build when TS version is locked to 2.8.4.  Any workaround on this ?

Yes, I used the previous version for now.
The error I was getting is following : 
```
> tsc

node_modules/@types/ramda/src/$operation.d.ts(9,11): error TS2322: Type 'T' is not assignable to type 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/$operation.d.ts(11,11): error TS2322: Type 'U' is not assignable to type 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(26,23): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(29,23): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(31,70): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(34,70): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(45,23): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(53,40): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(60,40): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'PropertyKey' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
```

At the moment cannot give you clear answer regarding the 2.9 TS version, as it fails due to memory allocation (not sure whose issue is that, still investigating in this as well)

Have similar issue. I'm using TS 2.9.2, types/ramda#dist.

Version from commit f7dd77a0b93b903be41f1c28de488daa344ff6fb doesn't compile, with this errors (same as @vkosachev in https://github.com/types/npm-ramda/issues/400#issuecomment-406210253):
```
node_modules/@types/ramda/src/$operation.d.ts(9,11): error TS2322: Type 'T' is not assignable to type 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

node_modules/@types/ramda/src/$operation.d.ts(11,11): error TS2322: Type 'U' is not assignable to type 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

node_modules/@types/ramda/src/$types.d.ts(7,107): error TS2344: Type 'K' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

node_modules/@types/ramda/src/$types.d.ts(8,98): error TS2344: Type 'K' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
node_modules/@types/ramda/src/dissoc.d.ts(26,23): error TS2344: Type 'V' does not satisfy the constraint 'string'.
    Type 'number' is not assignable to type 'string'.

node_modules/@types/ramda/src/dissoc.d.ts(29,23): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

node_modules/@types/ramda/src/dissoc.d.ts(31,70): error TS2344: Type 'V' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

.....
```

I also have `"keyofStringsOnly": true` option in tsconfig.json, if that matters

--- a/templates/$operation.d.ts
+++ b/templates/$operation.d.ts
@@ -1,4 +1,4 @@
-import { Morphism } from './$types';
+import { Morphism, PropKey } from './$types';
 
 export type DeepPartial<T> = { [K in keyof T]?: DeepPartial<T[K]> };
 
@@ -6,13 +6,11 @@ export type DeepPartial<T> = { [K in keyof T]?: DeepPartial<T[K]> };
 export type Evolver<T> = Morphism<T, T> | { [K in keyof T]?: Evolver<T[K]> };
 
 // from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
-export type Diff<T extends PropertyKey, U extends PropertyKey> = ({
-  [P in T]: P
-} &
+export type Diff<T extends PropKey, U extends PropKey> = ({ [P in T]: P } &
   { [P in U]: never } & { [x: string]: never })[T];
-export type Omit<T, K extends PropertyKey> = Pick<T, Diff<keyof T, K>>;
+export type Omit<T, K extends PropKey> = Pick<T, Diff<keyof T, K>>;
 
-export type Same<T extends PropertyKey, U extends PropertyKey> = Diff<
+export type Same<T extends PropKey, U extends PropKey> = Diff<
   T | U,
   Diff<T, U> | Diff<U, T>
 >;

--- a/templates/$types.d.ts
+++ b/templates/$types.d.ts
@@ -2,6 +2,7 @@
 
 export type Property = string | number | symbol;
 export type Path = List<Property>;
+export type PropKey = keyof any; // for backward compatibility (keyofStringsOnly)
 
 // general
 
@@ -14,12 +15,12 @@ export type IndexedListMorphism<T, U> = (
   index: number,
   list: List<T>,
 ) => U;
-export type IndexedObjectMorphism<T, U, K extends PropertyKey> = (
+export type IndexedObjectMorphism<T, U, K extends PropKey> = (
   value: T,
   index: number,
   object: Record<K, T>,
 ) => U;
-export type KeyedObjectMorphism<T, U, K extends PropertyKey> = (
+export type KeyedObjectMorphism<T, U, K extends PropKey> = (
   value: T,
   key: K,
   object: Record<K, T>,

--- a/templates/dissoc.d.ts
+++ b/templates/dissoc.d.ts
@@ -1,11 +1,11 @@
 import { Omit } from './$operation';
-import { Property } from './$types';
+import { Property, PropKey } from './$types';
 
 export function $keyof<T extends object, K extends keyof T>(
   property: K,
   object: T,
 ): Omit<T, K>;
-export function $record<U extends Record<V, any>, V extends PropertyKey>(
+export function $record<U extends Record<V, any>, V extends PropKey>(
   property: V,
   object: U,
 ): Omit<U, V>;

